ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,102 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+class RedHatAuthTool(object):
+    auth_tool = None
+
+    @classmethod
+    def getInstance(cls):
```

Comment:
Why is it necessary to turn the ``RedHatAuthSelect`` object into a singleton? Is it so expensive to create the instance or does the instance have to retain state across calls?

Reply:
It is not expensive at all. However, there were ideas to select an appropriate tool whenever authselect/authconfig is called. Authconfig can accumulate all the information and then be called just once. Authselect does not need this. When we make final decision to drop authconfig tool and cleanup the code this could be reworked as well.

Reply:
I stiill don't understand the need to have this as a singleton. Just create the instance of whatever you need.

Also, we're not usually using camel casing for methods.